--- configure.orig	Sun Jul  2 18:48:08 2006
+++ configure	Sat Dec  1 12:09:06 2007
@@ -4824,19 +4824,19 @@
 echo "${ECHO_T}no" >&6
 fi
 
-if test x$PKG_CONFIG = xno ; then
-  { { echo "$as_me:$LINENO: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&5
-echo "$as_me: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&2;}
-   { (exit 1); exit 1; }; }
-fi
+#if test x$PKG_CONFIG = xno ; then
+#  { { echo "$as_me:$LINENO: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&5
+#echo "$as_me: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&2;}
+#   { (exit 1); exit 1; }; }
+#fi
 
-if $PKG_CONFIG --atleast-pkgconfig-version 0.14 ; then
-  :
-else
-  { { echo "$as_me:$LINENO: error: *** pkg-config too old; version 0.14 or better required." >&5
-echo "$as_me: error: *** pkg-config too old; version 0.14 or better required." >&2;}
-   { (exit 1); exit 1; }; }
-fi
+#if $PKG_CONFIG --atleast-pkgconfig-version 0.14 ; then
+#  :
+#else
+#  { { echo "$as_me:$LINENO: error: *** pkg-config too old; version 0.14 or better required." >&5
+#echo "$as_me: error: *** pkg-config too old; version 0.14 or better required." >&2;}
+#   { (exit 1); exit 1; }; }
+#fi
 
 if test "x$enable_debug" = "xyes"; then
   if test x$cflags_set != xset ; then
