Index: mplayer.c
===================================================================
--- mplayer.c	(revisión: 27884)
+++ mplayer.c	(copia de trabajo)
@@ -724,7 +724,17 @@
 
 
   if(edl_records != NULL) free(edl_records); // free mem allocated for EDL
-  if(how) mp_msg(MSGT_CPLAYER,MSGL_INFO,MSGTR_ExitingHow,how);
+  if(how) {
+    mp_msg(MSGT_CPLAYER,MSGL_INFO,MSGTR_ExitingHow,how);
+    if(how==MSGTR_Exit_quit)
+      mp_msg(MSGT_IDENTIFY, MSGL_INFO, "ID_EXIT=QUIT\n");
+    else
+    if(how==MSGTR_Exit_error)
+      mp_msg(MSGT_IDENTIFY, MSGL_INFO, "ID_EXIT=ERROR\n");
+    else
+    if(how==MSGTR_Exit_eof) 
+      mp_msg(MSGT_IDENTIFY, MSGL_INFO, "ID_EXIT=EOF\n");
+  }
   mp_msg(MSGT_CPLAYER,MSGL_DBG2,"max framesize was %d bytes\n",max_framesize);
 
   exit(rc);
