Index: input/input.c
===================================================================
--- input/input.c	(revisión: 29930)
+++ input/input.c	(copia de trabajo)
@@ -172,6 +172,7 @@
   { MP_CMD_LOADLIST, "loadlist", 1, { {MP_CMD_ARG_STRING, {0}}, {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
   { MP_CMD_RUN, "run", 1, { {MP_CMD_ARG_STRING,{0}}, {-1,{0}} } },
   { MP_CMD_VF_CHANGE_RECTANGLE, "change_rectangle", 2, { {MP_CMD_ARG_INT,{0}}, {MP_CMD_ARG_INT,{0}}, {-1,{0}}}},
+  { MP_CMD_AF_EQ_SET, "af_eq_set_bands", 1, { {MP_CMD_ARG_STRING, {0}}, {-1,{0}}}}, //turbos
   { MP_CMD_TV_TELETEXT_ADD_DEC, "teletext_add_dec", 1, { {MP_CMD_ARG_STRING,{0}}, {-1,{0}} } },
   { MP_CMD_TV_TELETEXT_GO_LINK, "teletext_go_link", 1, { {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
 
Index: input/input.h
===================================================================
--- input/input.h	(revisión: 29930)
+++ input/input.h	(copia de trabajo)
@@ -129,6 +129,7 @@
   MP_CMD_ASS_USE_MARGINS,
   MP_CMD_SWITCH_TITLE,
   MP_CMD_STOP,
+  MP_CMD_AF_EQ_SET,
 
   /// DVDNAV commands
   MP_CMD_DVDNAV_UP = 1000,
Index: DOCS/tech/slave.txt
===================================================================
--- DOCS/tech/slave.txt	(revisión: 29930)
+++ DOCS/tech/slave.txt	(copia de trabajo)
@@ -472,7 +472,12 @@
     Run <value> as shell command. In OSD menu console mode stdout and stdin
     are through the video output driver.
 
+af_eq_set_bands <values>
+    Set bands gain in audio equalizer. Se details in manual (-af equalizer).
+    Example:
+       af_eq_set_bands 11:11:10:5:0:-12:0:5:12:12
 
+
 Available properties:
 
 name               type      min     max     get set step comment
Index: command.c
===================================================================
--- command.c	(revisión: 29930)
+++ command.c	(copia de trabajo)
@@ -2957,6 +2957,13 @@
 	    }
 	    break;
 
+	case MP_CMD_AF_EQ_SET:{  
+      af_instance_t* m1=af_get(sh_audio->afilter, "equalizer");
+      if (m1) m1->control( m1, AF_CONTROL_COMMAND_LINE, cmd->args[0].v.s);
+      else mp_msg(MSGT_CPLAYER, MSGL_INFO, "failed (forgot -af equalizer=0:0 ?)\n");
+	    }
+      break;
+
 	case MP_CMD_VF_CHANGE_RECTANGLE:
             if (!sh_video)
                 break;
Index: libaf/af_equalizer.c
===================================================================
--- libaf/af_equalizer.c	(revisión: 29930)
+++ libaf/af_equalizer.c	(copia de trabajo)
@@ -140,6 +140,7 @@
   case AF_CONTROL_COMMAND_LINE:{
     float g[10]={0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
     int i,j;
+    float gain_factor2=0.0;
     sscanf((char*)arg,"%f:%f:%f:%f:%f:%f:%f:%f:%f:%f", &g[0], &g[1],
 	   &g[2], &g[3], &g[4], &g[5], &g[6], &g[7], &g[8] ,&g[9]);
     for(i=0;i<AF_NCH;i++){
@@ -148,6 +149,17 @@
 	  pow(10.0,clamp(g[j],G_MIN,G_MAX)/20.0)-1.0;
       }
     }
+    
+    //calculate gainfactor for slave command
+    //see: af_eq_set_bands  -> input.c
+    //see code: AF_CONTROL_REINIT  in this file
+     for(j=0;j<KM;j++)
+        if(gain_factor2< s->g[0][j]) gain_factor2=s->g[0][j];
+     gain_factor2=log10(gain_factor2 + 1.0) * 20.0;
+     if(gain_factor2 > 0.0)   gain_factor2=0.1+(gain_factor2/12.0);
+     else gain_factor2=1;
+     s->gain_factor=gain_factor2;
+    
     return AF_OK;
   }
   case AF_CONTROL_EQUALIZER_GAIN | AF_CONTROL_SET:{
