Index: create_deb.sh
===================================================================
--- create_deb.sh	(revisión: 0)
+++ create_deb.sh	(revisión: 0)
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+svn_revision=`LC_ALL=C svn info 2> /dev/null | grep Revision | cut -d' ' -f2`
+test $svn_revision || svn_revision=`grep revision .svn/entries 2>/dev/null | cut -d '"' -f2`
+test $svn_revision || svn_revision=`sed -n -e '/^dir$/{n;p;q;}' .svn/entries 2>/dev/null`
+test $svn_revision || svn_revision=UNKNOWN
+
+dch -v "2:1.0rc2svn$svn_revision" "New version"
+
+#DEB_BUILD_OPTIONS="--enable-runtime-cpudetection --enable-gui" fakeroot debian/rules binary
+DEB_BUILD_OPTIONS="--enable-gui" MAKE_OPTS="-j 2" fakeroot debian/rules binary

Cambios de propiedades en create_deb.sh
___________________________________________________________________
Nombre: svn:executable
   + *

Index: debian/rules
===================================================================
--- debian/rules	(revisión: 27531)
+++ debian/rules	(copia de trabajo)
@@ -28,7 +28,7 @@
 	dh_testdir
 
 	# commands to compile the package.
-	${MAKE} 
+	${MAKE} $(MAKE_OPTS)
 
 	touch build-stamp
 
