Index: debian/control
===================================================================
--- debian/control	(revisión: 27554)
+++ debian/control	(copia de trabajo)
@@ -7,7 +7,11 @@
 
 Package: mplayer
 Architecture: any
-Depends: ${shlibs:Depends},debconf,libconfhelper-perl
+Depends: ${shlibs:Depends}, mplayer-skin-blue, ttf-bitstream-vera
+Replaces: mplayer-nogui, mplayer-doc, mencoder
+Conflicts: mplayer-nogui, mplayer-doc, mencoder
+Provides: mencoder, mplayer-doc
+Suggests: w32codecs
 Description: The Ultimate Movie Player 
  MPlayer is a movie player for LINUX (runs on many other Unices, and non-x86
  CPUs, see the ports section). It plays most MPEG, VOB, AVI, OGG/OGM, VIVO,
