Index: cfg-mencoder.h
===================================================================
--- cfg-mencoder.h	(revision 24928)
+++ cfg-mencoder.h	(working copy)
@@ -242,6 +242,9 @@
 	
 	{"odml", &write_odml, CONF_TYPE_FLAG, CONF_GLOBAL, 0, 1, NULL},
 	{"noodml", &write_odml, CONF_TYPE_FLAG, CONF_GLOBAL, 1, 0, NULL},
+
+	{"keep-pts", &keep_pts, CONF_TYPE_FLAG, 0, 0, 1, NULL},
+	{"nokeep-pts", &keep_pts, CONF_TYPE_FLAG, 0, 1, 0, NULL},
 	
 	// info header strings
 	{"info", info_conf, CONF_TYPE_SUBCONFIG, CONF_GLOBAL, 0, 0, NULL},
Index: DOCS/man/en/mplayer.1
===================================================================
--- DOCS/man/en/mplayer.1	(revision 24928)
+++ DOCS/man/en/mplayer.1	(working copy)
@@ -7068,6 +7068,14 @@
 Do not write OpenDML index for AVI files >1GB.
 .
 .TP
+.B \-keep\-pts
+Send the original presentation timestamp (PTS) down the filter and encoder
+chain.
+This may cause incorrect output ("badly interleaved") if the original PTS
+are wrong or the framerate is changed, but can be necessary for certain
+filters (such as ASS).
+.
+.TP
 .B \-noskip
 Do not skip frames.
 .
Index: mencoder.c
===================================================================
--- mencoder.c	(revision 24928)
+++ mencoder.c	(working copy)
@@ -204,6 +204,7 @@
 
 int auto_expand=1;
 int encode_duplicates=1;
+int keep_pts=0;
 
 // infos are empty by default
 char *info_name=NULL;
@@ -1316,7 +1317,8 @@
     // decode_video will callback down to ve_*.c encoders, through the video filters
     {void *decoded_frame = decode_video(sh_video,frame_data.start,frame_data.in_size,
       skip_flag>0 && (!sh_video->vfilter || ((vf_instance_t *)sh_video->vfilter)->control(sh_video->vfilter, VFCTRL_SKIP_NEXT_FRAME, 0) != CONTROL_TRUE), MP_NOPTS_VALUE);
-    blit_frame = decoded_frame && filter_video(sh_video, decoded_frame, MP_NOPTS_VALUE);}
+    blit_frame = decoded_frame && filter_video(sh_video, decoded_frame,
+	keep_pts ? sh_video->pts : MP_NOPTS_VALUE);}
     
     if (sh_video->vf_inited < 0) mencoder_exit(1, NULL);
     
